server/src/services/plugin.service.ts

```diff
@@ -0,0 +1,70 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { Plugin, PluginAction, PluginFilter } from 'src/database';
+import { PluginResponseDto } from 'src/dtos/plugin.dto';
+import { pluginTriggers } from 'src/schema/tables/plugin.table';
+import { BaseService } from 'src/services/base.service';
+
+@Injectable()
+export class PluginService extends BaseService {
+  async getAll(): Promise<PluginResponseDto[]> {
+    const plugins = await this.pluginRepository.getAllPlugins();
+    return Promise.all(plugins.map((plugin) => this.mapPlugin(plugin)));
```

Comment:
No, we should load the plugin, and join to the filters and actions all within the repository and return the whole plugin, not just the fields from the plugin table.